src/sync/sync_user.cpp

```diff
@@ -177,7 +191,8 @@ void SyncUser::register_session(std::shared_ptr<SyncSession> session)
         case State::Active:
             // Immediately ask the session to come online.
             m_sessions[path] = session;
-            if (m_is_admin) {
+            // FIXME: `SyncUser`s shouldn't even wrap admin tokens; the bindings should do that.
+            if (!m_should_persist) {
```

Comment:
It feels a bit odd to have a flag that controls whether or not we should persist the user be what determine whether or not the token is treated as an admin token or not.